File: Swift/conversor.swift
func decimalParaBinario(_ decimal: Int) -> String {
    if decimal == 0 { return "0" }

    var numero = decimal
    var binario = ""
    while numero > 0 {
        binario = String(numero % 2) + binario
        numero /= 2
    }
    return binario
}
print("Digite um numero decimal: ", terminator: "")
if let linha = readLine(), let decimal = Int(linha) {
    print("Binario: \(decimalParaBinario(decimal))")
}

File: Ruby/conversor.rb
def decimal_para_binario(decimal)
  return "0" if decimal == 0

  binario = ""
  numero = decimal
  while numero > 0
    binario = (numero % 2).to_s + binario
    numero /= 2
  end
  binario
end

print "Digite um numero decimal: "
decimal = gets.chomp.to_i
puts "Binario: #{decimal_para_binario(decimal)}"
